src/main/java/edu/tamu/sage/model/BasicValueOp.java

```diff
@@ -0,0 +1,27 @@
+package edu.tamu.sage.model;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+
+import edu.tamu.sage.model.validation.BasicValueOpValidator;
+
+@Entity
+public abstract class BasicValueOp extends BasicOp {
+
+    @Column(nullable = true)
```

Comment:
Does this have to become nullable? We will lose the database constraint.

Reply:
The constraint was acting on all entries for all types of operators. Since the mapping operator doesn't use it I had to remove the constraint. 

The weaver validation is still in place for any class that inherits the BasicValueOp.

Reply:
The `MappingOp` extends `BasicOp` and should not act on this constraint. It is a table per concrete type. I would not want to lose the database constraint for all the other now `BasicValueOp` operators.

Reply:
Well, looks like we are using SINGLE_TABLE inheritance. Darn.

Reply:
So, we have no reasonable choice but to lose the database constraint. The only alternative I can think of is to serialize a map into a string value and deserialize it when retrieving it from the database. Can be done with AttributeConverter but then we would have to accept JSON in a column. Going to just resolve this as is.